corgidrp/measure_companions.py

```diff
@@ -0,0 +1,450 @@
+
+    # iii. Measure reference PSF counts. An off-axis calibration PSF of a star at some known 
+    #    separation where mask effects are negligible
+    reference_psf_counts, _ = measure_counts(reference_psf, phot_method, None, **phot_method_kwargs)
+
+    # iv. Compute the ratio of photometry_host_star/photometry_reference_PSF
+    host_psf_ratio = host_star_counts/reference_psf_counts
+
+    ### Parse companion positions from headers
+    companions = [{"id": key, "x": float(x), "y": float(y)}
+                  for key, val in psf_sub_image.ext_hdr.items() if key.startswith('SNYX')
+                  for _, x, y in [val.split(',')]]
+
+    if verbose: print(f"Found {len(companions)} companion(s) in header.")
+
+    ### Loop over companions 
+    results = []
+    for comp in companions:
+        x_c, y_c = comp['x'], comp['y']
+
+        # v. Get/interpolate the off-axis PSF at the separation of the companion
+        throughput_factor, nearest_frame = measure_core_throughput_at_location(x_c, y_c, ct_cal, 
+                                                                               ct_dataset, FpamFsamCal)
+        
+        # vi. Scale the off-axis PSF by the computed ratio above to get the off-axis PSF of the host 
+        # star if it was at the same position as the companion
+        scaled_psf_data = nearest_frame.data * host_psf_ratio  
+
+        ## Create a new Image object with the same headers and error maps (if applicable)
+        scaled_psf_at_companion_loc = Image(
+            data_or_filepath=scaled_psf_data, 
+            pri_hdr=nearest_frame.pri_hdr, 
+            ext_hdr=nearest_frame.ext_hdr, 
+            err=nearest_frame.err if hasattr(nearest_frame, 'err') else None
+            )
+
+        # vii. Do PSF subtraction for the scaled_psf_at_companion_location, either though forward modeling
+        # or through classical PSF subtraction
+        
+        if forward_model == True:
+            # Get star location from headers
+            x_star = scaled_psf_at_companion_loc.ext_hdr.get('STARLOCX', None)
+            y_star = scaled_psf_at_companion_loc.ext_hdr.get('STARLOCY', None)
+            guesssep = np.sqrt((x_c - x_star) ** 2 + (y_c - y_star) ** 2)
+            guesspa = np.degrees(np.arctan2(x_c - x_star, y_star - y_c)) % 360  # Ensures PA is [0, 360] degrees
+            psf_guess_counts, _ = measure_counts(scaled_psf_at_companion_loc, phot_method, None, 
+                                                                 **phot_method_kwargs)
+
+            print("checking star loc", x_star, y_star)
+
+            fit = forward_model_psf(coronagraphic_dataset, scaled_psf_at_companion_loc, guesssep, guesspa,
+                                    psf_guess_counts, outputdir=output_dir)
+            print("forward model fit", fit)
+        else:
+            psf_sub_star_at_companion_loc, efficiency = classical_psf_sub(scaled_psf_at_companion_loc)
```

Comment:
are we really doing classical PSF subtraction, or we are just passing in any psf-subtracted frame (`psf_sub_image`) and measuring the flux relative to the `psf_sub_star_at_companion_loc`? 

It's totally ok if we're not. Directly measuring the flux of the companion and ignoring algorithm throughput is what I was basically thinking for a first pass (and we could use Ell's algorithm throughput term that they are writing into the output of the L4 PSF-subtracted image).